### PR TITLE
Feature/voip

### PR DESCRIPTION
**Pull request by SeniorZhai**

### CODE REVIEW

File: app/src/main/java/one/mixin/android/extension/FileExtension.kt [hunk 1]
```diff
@@ -235,6 +235,10 @@ fun File.createAudioTemp(type: String, noMedia: Boolean = true): File {
     return newTempFile("Audio_$time", ".$type", noMedia)
 }
 
+fun File.createRTCTemp(conversationId: String, messageId: String): File {
```
Comment (SeniorZhai) [inline app/src/main/java/one/mixin/android/extension/FileExtension.kt]: Unused

File: app/src/main/java/one/mixin/android/webrtc/PeerConnectionClient.kt [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            remoteCandidateCache.clear()
+        }
+        if (remoteSdpCache != null && peerConnection!!.remoteDescription == null) {
+            peerConnection!!.setRemoteDescription(remoteSdpObserver, remoteSdpCache)
+            remoteSdpCache = null
+        }
+    }
+
+    private fun createAudioTrack(): AudioTrack {
+        val audioConstraints = MediaConstraints().apply {
```
Comment (SeniorZhai) [inline app/src/main/java/one/mixin/android/webrtc/PeerConnectionClient.kt]: Unused code

File: app/src/main/java/one/mixin/android/webrtc/receiver/IncomingCallReceiver.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            Log.w(TAG, "Telephony event does not contain number...")
+            return
+        }
+
+        if (intent.getStringExtra(TelephonyManager.EXTRA_STATE) != TelephonyManager.EXTRA_STATE_RINGING) {
+            Log.w(TAG, "Telephony event is not state ringing...")
+            return
+        }
+        val callListener = InCallListener(context, Handler())
+        //TODO
```
Comment (SeniorZhai) [inline app/src/main/java/one/mixin/android/webrtc/receiver/IncomingCallReceiver.kt]: Is complete?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -469,5 +483,22 @@
     <!--Pending-->
     <string name="pending">Pending</string>
     <string name="pending_confirmations">%1$d/%2$d confimations</string>
+    <!-- webrtc -->
+    <string name="call_notification_outgoing">Calling...</string>
```
Comment (SeniorZhai) [inline app/src/main/res/values/strings.xml]: Replace `&#8230;`

**Comment by Tougee** [comment]
All fixed

**Pull request merged**